src/dbnode/persist/fs/msgpack/decoder_test.go

```diff
@@ -254,3 +254,25 @@ func TestDecodeBytesAllocNew(t *testing.T) {
 	}
 	require.Equal(t, []byte("testIndexEntry"), res.ID)
 }
+
+func TestDecodeIndexEntryInvalidChecksum(t *testing.T) {
+	var (
+		enc = NewEncoder()
+		dec = NewDecoder(nil)
+	)
+	require.NoError(t, enc.EncodeIndexEntry(testIndexEntry))
+
+	// Update to invalid checksum
+	enc.buf.Truncate(len(enc.Bytes()) - 5)
```

Comment:
Isn’t the checksum 8 bytes?

Reply:
Ahh, yeah, I need to update the design doc. We use the adler-32 algorithm which generates 4 byte checksums. Didn't know that when I wrote the initial doc. We get 5 here from the fact that msgpack prefixes a one byte integer code so that it knows how to decode the upcoming integer. I'll add a comment here to clarify.